### DISCUSSION TITLE
Concluding discussions

### DISCUSSION THREAD

**Discussion by JonathanGregory** [cf-convention/discuss#313] [Comments and ideas for changing CF]
### Topic for discussion

I wonder how our GitHub discussions should be dealt with when they stop. Some of them will be aware of coming to a conclusion, and others will just go quiet (like issues), I suspect. Here are a couple of questions:

* Should we _close_ them when they've stopped (for whatever reason) or leave them open? I think the only difference this makes is whether they appear in the default list. It's sorted by latest activity, so the inactive ones will gradually go down the list. With issues, there's a reason to close them, namely that the list of open ones compose our work agenda, but I don't think that argument applies as much to discussions.

* Should we label them, as we do with issues, e.g. as _concluded_ or _inconclusive_, if they are conscious of having reached an end of some sort? We could also label them automatically as _dormant_ if they have been quiet for some time without being explicitly concluded.

Best wishes

Jonathan

**Comment by DocOtak** [comment]
Repeating here a [comment I made in another question](https://github.com/orgs/cf-convention/discussions/324#discussioncomment-9704630):

>Re what to do about this open discussion, I thought there was a way to mark an answer and "resolve" the discussion, but I don't see that option for me. perhaps you @ChrisBarker-NOAA as the question asker can do that? https://github.blog/changelog/2023-04-17-marking-a-threaded-comment-as-the-answer/
>
>Marking something as the answer will pin it under the question in "green" so folks who find this question again can jump straight to the answer.

**Reply by DocOtak** [reply]
@JonathanGregory when setting discussion categories to allow things to be "marked as answer" it the category needs to change to "Question / Answer"

<img width="760" alt="Screenshot 2024-06-07 at 09 33 51" src="https://github.com/cf-convention/discuss/assets/868027/2ffd2206-c0d7-47c5-9a9f-ca001ac8a354">

**Comment by JonathanGregory** [comment]
Thanks, @DocOtak. If you "mark as answer" does it close the question? I assume that anyone who opens a discussion can change its category subsequently (e.g. comment to question).

**Reply by DocOtak** [reply]
Looks like it adds a green checkbox next to the topic and pins the answer directly under the question, see https://github.com/tiangolo/fastapi/discussions/categories/questions for some active Q/A style discussions. The "Discussion format" option is global for that category, I think it would be appropriate for our QA category.

**Comment by ChrisBarker-NOAA** [comment]
Hmm -- for #324 -- I only see "close" as an option, no "mark as answer" -- and the catagory is already "Q&A about using CF".

**Reply by DocOtak** [reply]
It's the "discussion format" option that would need to change (the radio buttons in my screenshot)

**Comment by ChrisBarker-NOAA** [comment]
Ahh -- so that's a global (to the Catagory) setting, not a per-topic setting.

So I'll let one of you change it if you see fit.

**Comment by JonathanGregory** [comment]
There are four types of Discussion format offered by GitHub: _open-ended discussions_, _announcements_, _questions/answers_, and _polls_. I created our three categories (**announcements**, **comments** and **Q&A**) all with the _open-ended discussion_ format. This is the natural choice for **comments**. The sole difference between _announcement_ format and _open-ended discussion_ format is that discussions with the _announcement_ format can be started only by a maintainer, not by anyone, and I didn't think we needed that restriction on our **announcements**. I didn't create a _polls_ category because we've always decided things by consensus, not voting. We could introduce it of course if it were needed.

I guess that "Mark as answered" is supported by the _questions/answers_ format. I didn't choose that format for our **Q&A** because it has the upvoting facility, like you see e.g. in stackexchange. I don't think our **Q&A** are like that. Generally there are few answers, and often a discussion is needed to arrive at the right answer. Maybe we should use the _questions/answers_ format. What do you think? I tend to think that it's easier to keep the same format, and introduce a label instead to "Mark as answered". We're accustomed to using labels effectively, I would say.

**Comment by larsbarring** [comment]
I have just been looking through the list of discussions and there are several that resulted in an issue that has been implemented, or otherwise are of little interest to keep open. I suggest they should be closed to avoid getting an ever expanding list of discussion that eventually become unmanageable. I suggest the following can be closed (no particular order):

https://github.com/orgs/cf-convention/discussions/300
https://github.com/orgs/cf-convention/discussions/362
https://github.com/orgs/cf-convention/discussions/320
https://github.com/orgs/cf-convention/discussions/339
https://github.com/orgs/cf-convention/discussions/322
https://github.com/orgs/cf-convention/discussions/310
https://github.com/orgs/cf-convention/discussions/342

This is a suggestion and others might want to add to this list, or remove from it.

**Comment by taylor13** [comment]
I'm not sure what the status is of the pull request mentioned in [this comment](https://github.com/orgs/cf-convention/discussions/322), but until that is acted on I think we should probably keep. comment 322 open for now.  I would not object to closing the others.

**Comment by ChrisBarker-NOAA** [comment]
> I'm not sure what the status is of the pull request mentioned in this
> comment <https://github.com/orgs/cf-convention/discussions/322>,
>

Looks like it’s been merged.

**Reply by JonathanGregory** [reply]
Lars has [confirmed](https://github.com/orgs/cf-convention/discussions/322#discussioncomment-11510063) that it has been merged. It's about the page on [Projects and Activities that Use the CF Metadata Conventions](https://cfconventions.org/projects-activities.html), which now exists.

**Comment by JonathanGregory** [comment]
In Lars's list, [362](https://github.com/orgs/cf-convention/discussions/362) and [339](https://github.com/orgs/cf-convention/discussions/339) are Q&A-type. I think there's a value in keeping Q&As open so that they will be noticed by someone looking for answers. They could be linked to answers in the FAQ as well.

The others are of comment/idea-type. I agree they could be closed. The most recent update to any of those was on 23rd August, 3.5 months ago. Could we adopt a guideline of closing any discussion if it has been quiet for more than three months? Of course, if it has been clearly concluded, it could be closed earlier - or is there any value in keeping it open for a while after being concluded?

**Comment by ChrisBarker-NOAA** [comment]
> In Lars's list, [362](https://github.com/orgs/cf-convention/discussions/362) and [339](https://github.com/orgs/cf-convention/discussions/339) are Q&A-type. I think there's a value in keeping Q&As open so that they will be noticed by someone looking for answers.

Does closing them make them harder to search / find? If so then yes, let's keep them open, but that's too bad. "asked and answered" would be a good category to have and be easy to search.

> They could be linked to answers in the FAQ as well. -- good idea, and then they could be closed.

>  Could we adopt a guideline of closing any discussion if it has been quiet for more than three months?

I don't think that's a good idea, ideas can sleep longer than that and still be useful -- a year, if at all?

I don't know how hard it is to set up, but some projects have a bot that sends out an email to issue particpants (hopefully dicussions as well) saying that an issue has been dormant for some time, and that it will be closed if no one revives it -- that would be4 a good idea -- anyone have the github-fu to make that happen?

[in the meantime, I suppose we could do it by hand]

**Comment by ChrisBarker-NOAA** [comment]
Here's the status, as I see it:

I took it upon myself to close a few that seemed clearly ready to be closed -- they can always be re-opened if I made a mistake:


https://github.com/orgs/cf-convention/discussions/300: concluded -- I've closed it.

https://github.com/orgs/cf-convention/discussions/320: "Broken web links" -- This looks not totally resolved, but should probably be turned into an issue for any remaining ones. -- I posted to that discussion asking for resolution

https://github.com/orgs/cf-convention/discussions/322: this has been resolved -- more to do to improve it (links to CF projects) -- but I don't think we need it anymore. -- I just closed it, but someone can re-open if they think that it needs more attention.

https://github.com/orgs/cf-convention/discussions/310: I think this is done, but leaving it to the standard_name gurus to close it.

https://github.com/orgs/cf-convention/discussions/342: this is about the 2024 CF workshop -- closed it.

### Questions:

https://github.com/orgs/cf-convention/discussions/339: question: keep open until / if a FAQ entry is added

https://github.com/orgs/cf-convention/discussions/362: question: keep open until / if a FAQ entry is added
